src/services/vehicleDragger.ts

```diff
@@ -84,25 +112,33 @@ const enum DragState
 	Cancel
 }
 
+interface DragPosition
+{
+	tilePosition: CoordsXYZ,
+	trackPosition: CarTrackLocation | null,
+	trackProgress: number | null,
+}
+
 /**
  * Arguments of a currently dragged vehicle.
  */
 interface DragVehicleArgs
 {
 	target: number;
-	position: CoordsXYZ;
+	position: DragPosition;
 	state: DragState;
 }
 
 /**
  * Get a possible position to drag the vehicle to.
  */
-function getPositionFromTool(args: ToolEventArgs, vehicleType: RideObjectVehicle | null): CoordsXYZ | null
+function getPositionFromTool(args: ToolEventArgs, vehicleType: RideObjectVehicle | null, currentId: number): [CoordsXYZ | null, CarTrackLocation | null]
```

Comment:
Wouldn't it be better to make this return type a single object with all the required fields, instead of a tuple with two different objects and some duplicate fields? 😅

Reply:
sure will fix.